corehq/util/hmac_request.py

```diff
@@ -14,6 +14,16 @@
 _soft_assert = soft_assert(notify_admins=True)
 
 
+def convert_to_bytestring_if_unicode(shared_key):
+    return shared_key.encode('utf-8') if isinstance(shared_key, six.text_type) else shared_key
+
+
+def get_hmac_digest(shared_key, data):
+    hm = hmac.new(shared_key, data, hashlib.sha256)
```

Comment:
since you're always going to want a bytestring here you could put the call to `convert_to_bytestring_if_unicode` in this function which would avoid having to call it each time the function is called

Reply:
updated https://github.com/dimagi/commcare-hq/pull/20916/commits/2e62ff7b0500a4f4731eb82856f171ff49192d46